### PR TITLE
Alternative options for creating a serial copy of alpaka modules

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

File: HeterogeneousCore/AlpakaCore/python/functions.py
Comment (fwyzard) [inline HeterogeneousCore/AlpakaCore/python/functions.py:1]: @makortel what do you think ?

Comment (makortel) [reply HeterogeneousCore/AlpakaCore/python/functions.py:1]: I think this is much better than spreading around the direct modification `copy._TypedParameterizable__type`, and a reasonable way for immediate needs. For the longer term I'd still like to solve the problem in different way, with which I'll follow up in https://github.com/cms-sw/cmssw/issues/43780

File: HeterogeneousCore/AlpakaCore/python/functions.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+def makeSerialClone(module, **kwargs):
+    type = module._TypedParameterizable__type
+    if type.endswith('@alpaka'):
+        # alpaka module with automatic backend selection
+        base = type.removesuffix('@alpaka')
+    elif type.startswith('alpaka_serial_sync::'):
+        # alpaka module with explicit serial_sync backend
+        base = type.removeprefix('alpaka_serial_sync::')
```
Comment (makortel) [inline HeterogeneousCore/AlpakaCore/python/functions.py:8]: The `alpaka_serial_sync` case could just `return module.clone(**kwargs)`.

File: HeterogeneousCore/AlpakaCore/python/functions.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    elif type.startswith('alpaka_cuda_async::'):
+        # alpaka module with explicit cuda_async backend
+        base = type.removeprefix('alpaka_cuda_async::')
+    elif type.startswith('alpaka_rocm_async::'):
+        # alpaka module with explicit rocm_async backend
+        base = type.removeprefix('alpaka_rocm_async::')
```
Comment (makortel) [inline HeterogeneousCore/AlpakaCore/python/functions.py:14]: The if-else chain could be replaced with e.g. a regex to remove the `alpaka_[^:]*::` part.

**Pull request closed without merging**